### PR TITLE
Saving the generated artist agreement after minting is requested

### PR DESCRIPTION
**Pull request by escobarjonatan**

### CODE REVIEW

**Comment by dmkirshon** [comment]
Just to verify, the artist agreement itself is still not generating? Even with company field filled in?
Screenshot from 2023-04-13 10-42-01

File: src/pages/home/uploadSong/UploadSong.tsx [hunk 1]
```diff
@@ -84,6 +83,16 @@ const UploadSong: FunctionComponent = () => {
...
+    generateArtistAgreement({
+      body: {
+        songName: values.title,
+        companyName,
+        artistName,
+        stageName,
+        songId: id,
+        save: true,
+      },
+    });
```
Comment (dmkirshon) [inline src/pages/home/uploadSong/UploadSong.tsx:95]: So even if the person doesn't select minting (conditional in `handleSongInfo`) the agreement gets generated? or is this part of the initial backend testing?

Comment (escobarjonatan) [reply src/pages/home/uploadSong/UploadSong.tsx]: Good catch, forgot to wrap it in a condition :)

**Comment by escobarjonatan** [comment]
@dmkirshon The PDF generation is currently not working currently, @scandycuz can shine more light on the reasons? 👀

File: src/pages/home/uploadSong/UploadSong.tsx [hunk 2]
```diff
@@ -84,7 +83,21 @@ const UploadSong: FunctionComponent = () => {
...
   const handleSubmit = (values: UploadSongRequest) => {
-    uploadSong(values);
+    if (values.isMinting) {
+      generateArtistAgreement({
+        body: {
+          songName: values.title,
+          companyName,
+          artistName,
+          stageName,
+          songId: id,
```
[the remark this answers is not in the document]
Comment (escobarjonatan) [reply src/pages/home/uploadSong/UploadSong.tsx]: uhm this is awk lol i knew i should have done the 2 at the same time. I have updated the upload song thunk

**Pull request merged**